code/workspaces/infrastructure-api/resources/vscode.deployment.template.yml

```diff
@@ -0,0 +1,106 @@
+spec:
+  selector:
+    matchLabels:
+      name: {{ name }}
+  template:
+    metadata:
+      labels:
+        name: {{ name }}
+        user-pod: {{ type }}
+    spec:
+      securityContext:
+        runAsUser: 0
+      serviceAccountName: {{ serviceAccount }}
+      initContainers:
+        # This container will generate the expected folder structure, which has a workspace folder for vscode and conda folder for conda.
+        # As the container runs a root these directories are owned by root and can
+        # not be written to by datalabs:users (user:group; 1000:100).
+        - name: create-notebook-folders
+          image: busybox
+          imagePullPolicy: IfNotPresent
+          command: ["sh"]
+          args: ["-c", "mkdir -p /mnt/persistentfs/vscode/{{ name }}/workspace && mkdir -p /mnt/persistentfs/vscode/{{ name }}/conda"]
+          volumeMounts:
+            - mountPath: /mnt/persistentfs
+              name: persistentfsvol
+        # This container sets the folder permissions for the generated workspace and conda directories to be usable by
+        # datalabs:users.
+        - name: set-directory-permissions
+          image: busybox
+          imagePullPolicy: IfNotPresent
+          command: ["sh"]
+          args: ["-c", "chmod 777 /mnt/persistentfs/notebooks && chmod 777 /mnt/persistentfs/vscode/{{ name }} && chmod 777 /mnt/persistentfs/vscode/{{ name }}/workspace && chmod 777 /mnt/persistentfs/vscode/{{ name }}/conda && ln -sfn /assets /mnt/persistentfs/vscode/{{ name }}/assets"]
+          volumeMounts:
+            - mountPath: /mnt/persistentfs
+              name: persistentfsvol
+      containers:
+        - image: {{ &vscode.image }}
+          imagePullPolicy : "IfNotPresent"
+          name: {{ name }}
+          command: ["/bin/sh",
+            "-c",
+            "exec /home/.openvscode-server/bin/openvscode-server --host 0.0.0.0 --connection-token $(ACCESS_TOKEN)"
+          ]
+          ports:
+            - containerPort: 3000
+              protocol: TCP
+          env:
+            - name: ACCESS_TOKEN
+              valueFrom:
+                secretKeyRef:
+                  name: {{ name }}
+                  key: token
+            - name: GRANT_SUDO
+              value: "{{ grantSudo }}"
+            # - name: JUPYTER_DATA_DIR
+            #   value: "/data/.jupyter"
+            - name: CONDA_ENV_DIR
+              value: "/data/conda/"
+          resources:
+            requests:
+              cpu: 200m
+              memory: 128Mi
+            limits:
+              cpu: 2
+              memory: 8Gi
+          livenessProbe:
+            httpGet:
+              path: {{{ basePath }}}
+              port: 3000
+            initialDelaySeconds: 5
+            periodSeconds: 10
+          volumeMounts:
+            {{#volumeMount}}
+            - name: persistentfsvol
+              mountPath: /data
+            {{/volumeMount}}
+            # - name: spark-kubernetes-properties
```

Comment:
Could these comments be deleted (here and in `createVSCodeDeployment`)?

Reply:
Yes - i've removed all irrelevant comments.